sql/sql_statistics.cc

```diff
@@ -2660,12 +2660,32 @@ int collect_statistics_for_table(THD *thd, TABLE *table)
 
   if(file->ha_table_flags() & HA_NATIVE_SAMPLING)
```

Comment:
I really dislike this flag. It is really simple to make default implementation in handler class of sampling which reads everything (maybe discard some records) and efficient implementation where it is needed.

Reply:
I think it's up to optimizer to choose the sampling method. I was also thinking to add a switch later. Besides, it will make it easier to test and compare the results